backend/capellacollab/projects/toolmodels/modelsources/git/github/handler.py

```diff
@@ -60,16 +60,24 @@ def get_artifact_from_job_as_content(
         ).encode()
 
     def __get_file_from_repository(
-        self, project_id: str, trusted_file_path: str, headers: dict = None
+        self,
+        project_id: str,
+        trusted_file_path: str,
+        revision: str | None = None,
+        headers: dict | None = None,
     ) -> requests.Response:
+        branch = revision if revision else self.git_model.revision
         return requests.get(
-            f"{self.git_instance.api_url}/repos{project_id}/contents/{parse.quote(trusted_file_path, safe='')}?ref={parse.quote(self.git_model.revision, safe='')}",
+            f"{self.git_instance.api_url}/repos{project_id}/contents/{parse.quote(trusted_file_path)}?ref={parse.quote(branch, safe='')}",
```

Comment:
```suggestion
            f"{self.git_instance.api_url}/repos/{project_id}/contents/{parse.quote(trusted_file_path)}?ref={parse.quote(branch, safe='')}",
```

Without a separating `/`, the URL looks a bit weird. Instead, I'd change the project_id to be in the format `{owner}/{repo_name}` (remove the leading `/`).
It has to be updated in all references of the Github URL.